cpp/src/fil/internal.cuh

```diff
@@ -242,7 +242,7 @@ struct node_traits {
   using forest               = sparse_forest<node_t>;
   static const bool IS_DENSE = false;
   static const storage_type_t storage_type_enum =
-    std::is_same<sparse_node16<typename node_t::real_t>, node_t>() ? SPARSE : SPARSE8;
+    std::is_same<sparse_node16<real_t>, node_t>() ? SPARSE : SPARSE8;
```

Comment:
This seems compile-time to me? In that case I'd recommend using `constexpr` which guarantees substitution at compile-time

Reply:
Done (`static constexpr storage_type_enum`).